src/App.tsx

```diff
@@ -5,10 +5,18 @@ import Nav from './Nav';
 import CardGrid from './CardGrid';
 import DetailPage from './DetailPage';
 import pokeload from './pikapokeball.gif';
-import { AllPokemonAPI, MetaPokemon, PokemonAttributes } from './types/index';
+import SearchFilter from './SearchFilter';
+import {
+  AllPokemonAPI,
+  MetaPokemon,
+  PokemonAttributes,
+  PokemonFilter,
+} from './types/index';
 
 const App: FunctionComponent<{}> = () => {
   const [allPokemon, setAllPokemon] = useState<Array<PokemonAttributes>>([]);
+  const [search, setSearch] = useState<boolean>(false);
```

Comment:
I think the name could be better here. `isSearchVisible`, `canSearch`, `isSearchable`, `dealersChoice`, etc.